tests/unit/test_synapse_workload.py

```diff
@@ -67,55 +68,67 @@ def test_enable_metrics_error(monkeypatch: pytest.MonkeyPatch):
         synapse.enable_metrics(container_mock)
 
 
-def test_enable_saml_success(harness_with_saml: Harness, monkeypatch: pytest.MonkeyPatch):
+def test_enable_saml_success():
     """
     arrange: set mock container with file.
     act: change the configuration file.
     assert: new configuration file is pushed and SAML is enabled.
     """
-    harness = harness_with_saml
-    config_content = """
-    listeners:
-        - type: http
-          port: 8080
-          bind_addresses:
-            - "::"
-          x_forwarded: false
-    """
-    text_io_mock = io.StringIO(config_content)
-    pull_mock = Mock(return_value=text_io_mock)
-    push_mock = MagicMock()
-    container_mock = MagicMock()
-    monkeypatch.setattr(container_mock, "pull", pull_mock)
-    monkeypatch.setattr(container_mock, "push", push_mock)
-
-    synapse.enable_saml(container_mock, harness.charm._charm_state)
-
-    assert pull_mock.call_args[0][0] == SYNAPSE_CONFIG_PATH
-    assert push_mock.call_args[0][0] == SYNAPSE_CONFIG_PATH
-    saml_relation = harness.model.get_relation("saml")
-    assert saml_relation
-    saml_relation_data = harness.get_relation_data(saml_relation.id, "saml-integrator")
-    assert saml_relation_data
-    entity_id = saml_relation_data.get("entity_id")
-    metadata_url = saml_relation_data.get("metadata_url")
+    # This test was given as an example in this comment by Ben Hoyt.
```

Comment:
Does this need to be a comment in the test?

Reply:
Yes, at least until all the tests are refactored and this one is the reference. So anyone can see more details about why this one is different.